content-for/test-body-footer.html

```diff
@@ -14,14 +14,35 @@
     a11yCheckCallback: function(results) {
       var violations = results.violations;
 
-      for (var i = 0, l = violations.length; i < l; i++) {
-        Ember.Logger.error('Violation #' + (i + 1), violations[i]);
-      }
+      if (violations.length) {
+        Ember.Logger.error('ACCESSIBILITY VIOLATIONS: ' + violations.length);
+
+        for (var i = 0, l = violations.length; i < l; i++) {
+          var violation = violations[i];
+          var violationNodeString = axe.ember.nodesToString(violation.nodes);
 
-      Ember.assert('The page should have no accessibility violations.', !violations.length);
+          Ember.Logger.warn(violation.impact.toUpperCase() + ': ' + violation.help);
+          Ember.Logger.info('Offending markup (' + violation.nodes.length + '): ' + violationNodeString);
+          Ember.Logger.info('Additional info: ' + violation.helpUrl);
+          Ember.Logger.info('-------------------------------------');
+        }
+      }
+      Ember.assert('The page should have no accessibility violations. Please check the developer console for more details.', !violations.length);
     },
 
     /**
+     * Takes the violations' nodes param and converts it to
+     * a string that can be displayed with the console error.
+     * @param {Array} nodes
+     * @return {String}
+     */
+     nodesToString: function(nodes) {
+       return nodes.map(function(node) {
+         return node.html;
+       }).join(', ');
```

Comment:
Good point. There is potential for it to contain a lot of nodes, so perhaps the intent of having the node list at this level makes it less useful.

I can potentially log how many specific violations of that type are present and instead provide an object or array of the offending elements.


On Jun 2, 2016, 12:30 AM -0400, Nathan Hammond<notifications@github.com>, wrote:
>  
> Incontent-for/test-body-footer.html(https://github.com/ember-a11y/ember-a11y-testing/pull/25#discussion_r65482880):
>  
> >},>>/**>+   * Takes the violations' nodes param and converts it to>+   * a string that can be displayed with the console error.>+   * @param {Array} nodes>+   * @return {String}>+   */>+   nodesToString: function(nodes) {>+    return nodes.map(function(node) {>+     return node.html;>+    }).join(', ');
>  
> What if this string is absurdly long?
>  
>  
> —
> You are receiving this because you authored the thread.
> Reply to this email directly,view it on GitHub(https://github.com/ember-a11y/ember-a11y-testing/pull/25/files/f15c6f6bb86efa59014b009c10f876c3a0d8d07a#r65482880), ormute the thread(https://github.com/notifications/unsubscribe/AAjYQcgi1ArUeP2C8jfGzrYQVyhy-LiBks5qHlxrgaJpZM4IsIU9).
>  
>  
>